<!-- Launch file for Segway platform startup -->

<launch>
    <arg name="GripperID" default="1"/>
    <arg name="GripperModel" default="AG95_MB"/>
    <arg name="Connectport" default="/dev/ttyUSB0"/>
    <arg name="Baudrate" default="115200"/>


    <node pkg="dh_gripper_driver" type="dh_gripper_driver" name="dh_gripper_driver" respawn="true" output="screen">
        <param name="Gripper_ID" type="string" value="$(arg GripperID)" />
        <param name="Gripper_Model" type="string" value="$(arg GripperModel)" />
        <param name="Connect_port" type="string" value="$(arg Connectport)" />
        <param name="BaudRate"  type="string" value="$(arg Baudrate)" />
    </node>

    <node pkg="dh_gripper_action_server" type="dh_gripper_action_server" name="dh_gripper_action_server" output="screen">
        <param name="min_angle" type="double" value="0.0" />
        <param name="max_angle" type="double" value="0.93" />
        <param name="min_effort" type="double" value="20.0" />
        <param name="max_effort" type="double" value="100.0" />
        <param name="default_effort" type="double" value="100.0" />
        <param name="speed" type="double" value="100.0" />
        <param name="action_server_name" type="str" value="gripper_controller/gripper_cmd" />
        <param name="control_topic" type="str" value="gripper/ctrl" />
        <param name="state_topic" type="str" value="gripper/states" />
        <param name="joint_states_topic" type="str" value="joint_states" />
        <param name="joint_name" type="str" value="left_outer_knuckle_joint" />
    </node>

</launch>
